packages/web-app-files/tests/unit/views/PrivateLink.spec.js

```diff
@@ -1,16 +1,142 @@
+import { shallowMount } from '@vue/test-utils'
+import Vuex from 'vuex'
+import { localVue } from './views.setup'
+import PrivateLink from '@files/src/views/PrivateLink.vue'
+
+localVue.prototype.$client.files = {
+  getPathForFileId: jest.fn(() => Promise.resolve('/lorem.txt'))
+}
+
+const backgroundImgPath = 'assets/loginBackground.jpg'
+const loginLogoPath = 'assets/logo.svg'
+
+const $route = {
+  params: {
+    fileId: '2147491323'
+  },
+  meta: {
+    title: 'Resolving private link'
+  }
+}
+
+const selectors = {
+  backgroundImg: '[data-testid="files-private-link-bg"]',
+  pageTitle: '[data-testid="files-private-link-page-title"]',
+  loginLogo: '.oc-login-logo',
+  loader: '[data-testid="files-private-link-loader"]',
+  ocSpinner: 'oc-spinner-stub',
+  errorMessage: '[data-testid="files-private-link-error-msg"]'
+}
+
 describe('PrivateLink view', () => {
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
   describe('when the page has loaded successfully', () => {
-    it.todo('should have the background image set')
-    it.todo('should display the page title')
-    it.todo('should display the logo')
-    it.todo('should resolve the provided file id to a path')
+    let wrapper
+    beforeEach(() => {
+      wrapper = getShallowWrapper()
+    })
+
+    it('should have the background image set', () => {
+      const backgroundImg = wrapper.find(selectors.backgroundImg)
+
+      expect(backgroundImg.exists()).toBeTruthy()
+      expect(backgroundImg.attributes().style).toEqual(
+        `background-image: url(${backgroundImgPath});`
+      )
+    })
+    it('should display the page title', () => {
+      const pageTitle = wrapper.find(selectors.pageTitle)
+
+      expect(pageTitle.exists()).toBeTruthy()
```

Comment:
Would it be reasonable to omit the `data-testid`s for the success case and just use a snapshot test here? 🤔